executor/adapter.go

```diff
@@ -1480,15 +1480,37 @@ func resetCTEStorageMap(se sessionctx.Context) error {
 // LogSlowQuery is used to print the slow query in the log files.
 func (a *ExecStmt) LogSlowQuery(txnTS uint64, succ bool, hasMoreResults bool) {
 	sessVars := a.Ctx.GetSessionVars()
+	// Disable recording the internal queries.
+	if sessVars.ConnectionID == 0 {
+		return
+	}
 	stmtCtx := sessVars.StmtCtx
 	level := log.GetLevel()
 	cfg := config.GetGlobalConfig()
 	costTime := time.Since(sessVars.StartTime) + sessVars.DurationParse
 	threshold := time.Duration(atomic.LoadUint64(&cfg.Instance.SlowThreshold)) * time.Millisecond
+	expensiveRatio := atomic.LoadUint64(&cfg.Instance.ExpensiveScanRatio)
 	enable := cfg.Instance.EnableSlowLog.Load()
+
+	// Check if it's an expensive mvcc scan. It's regarded as expensive if
+	// 1. The rocksdb_delete_skipped_count > processed_keys * expensive_ratio
+	// 2. The processed_keys is 0 but the duration is larger than slow threshold.
+	execDetail := stmtCtx.GetExecDetails()
+	isExpensiveScan := false
```

Comment:
I'm afraid the word *expensive* is not expressive enough. By literal, I think it might mean queries that runs for a long time but not necessarily expensive for each single key by average.
How do you think about using the word *inefficient* instead of *expensive*?

Reply:
Refactored PTAL.